xdsl/dialects/emitc.py

```diff
@@ -302,6 +304,73 @@ def is_supported_emitc_type(type_attr: Attribute) -> bool:
             return False
 
 
+@irdl_op_definition
+class EmitC_AddOp(IRDLOperation):
+    """
+    Addition operation.
+
+    With the `emitc.add` operation the arithmetic operator + (addition) can
+    be applied. Supports pointer arithmetic where one operand is a pointer
+    and the other is an integer or opaque type.
+
+    Example:
+
+    ```mlir
+    // Custom form of the addition operation.
+    %0 = emitc.add %arg0, %arg1 : (i32, i32) -> i32
+    %1 = emitc.add %arg2, %arg3 : (!emitc.ptr<f32>, i32) -> !emitc.ptr<f32>
+    ```
+    ```c++
+    // Code emitted for the operations above.
+    int32_t v5 = v1 + v2;
+    float* v6 = v3 + v4;
+    ```
+    """
+
+    name = "emitc.add"
+
+    lhs = operand_def()
+    rhs = operand_def()
+    result = result_def()
+
+    assembly_format = "operands attr-dict `:` functional-type(operands, results)"
+
+    def __init__(
+        self,
+        lhs: SSAValue,
+        rhs: SSAValue,
+        result_type: Attribute | None = None,
+    ):
+        if result_type is None:
```

Comment:
on second thought I'm not sure if this is a good idea... I had implemented this because I thought it would be a nice feature but now I'm thinking it could hide mistakes from the user

Reply:
Yeah I think this is minor because the most important thing is the user has the flexibility to do the thing they want, but it definitely feels like we can't infer the correct type in general.

Reply:
yup, removed it :)